src/map/ai/ai_char_normal.cpp

```diff
@@ -1186,7 +1186,22 @@ void CAICharNormal::ActionRangedFinish()
         }
 
         // only remove detectables
-        m_PChar->StatusEffectContainer->DelStatusEffectsByFlag(EFFECTFLAG_DETECTABLE);
+		m_PChar->StatusEffectContainer->DelStatusEffectsByFlag(EFFECTFLAG_DETECTABLE);
+
+		// Handle stealth shot.
+		CStatusEffect* stealthShot = m_PChar->StatusEffectContainer->GetStatusEffect(EFFECT_STEALTH_SHOT);
+		if (stealthShot != nullptr) {
+
+			// Make sure that the target is a mob.
+			if (m_PBattleTarget->objtype == TYPE_MOB) {
+				uint16 power = stealthShot->GetPower();
+
+				((CMobEntity*)m_PBattleSubTarget)->PEnmityContainer->LowerEnmityByPercent(m_PChar, power, nullptr);

```

Comment:
I am not sure if I follow completely.  Are you suggesting adding a comment to describe why the statement is being executed, or something else?